common/setups/rasr/util/nn.py

```diff
@@ -188,10 +192,13 @@ def get_data_dict(self):
                     "class": "OggZipDataset",
                     "audio": self.audio,
                     "partition_epoch": self.partition_epoch,
-                    "path": tuple(self.oggzip_files.get_path()),
+                    "path": [
+                        oggzip_file.get_path() for oggzip_file in self.oggzip_files
```

Comment:
I see the potential that this would hard-code hashes?!
Can we avoid that?

Reply:
uhm.. not using `get_path()` here but leave in the `tk.Path` object s.t. it will be converted by the write config method?

Reply:
only one way to find out xP

Reply:
In the `ReturnnConfig` https://github.com/rwth-i6/i6_core/blob/e2cce78e2be5a9ad3f04cbb48d743b533aaf628b/returnn/config.py#L254

```
pp = pprint.PrettyPrinter(indent=2, width=150, **self.pprint_kwargs)
for k, v in sorted(config.items()) if self.sort_config else config.items():
    if pprint.isreadable(v):
        config_lines.append("%s = %s" % (k, pp.pformat(v)))
    else:
        unreadable_data[k] = v
```

`v` should be the OggZip dict object. The dict should get parsed as a string via `pp.pformat`. The default depth is None. Which should convert all nested objects as well. And the Sisyphus Path Object `__str__` method returns the cached path as a `str`. So should work... I think :D

Reply:
@vieting can you confirm? xP

Reply:
Yes, that works 👍